app/workers/project_branches_retriever.rb

```diff
@@ -0,0 +1,45 @@
+class ProjectBranchesRetriever
+  include Sidekiq::Worker
+
+  def perform(user_id, project_id)
+    @user_id = user_id
+    @project_id = project_id
+    return unless project.present?
+    update_branches
+  end
+
+  def update_branches
+    branches.each { |branch| create_or_update(branch) }
+  end
+
+  private
+
+  def create_or_update(github_branch)
+    branch = project.branches.find_or_create_by(name: github_branch[:name])
+    branch.update_attributes!(default: default?(branch.name))
```

Comment:
Why not leaving this default in false and only make the update if default?(branch_name) is true